src/spectral.ts

```diff
@@ -72,7 +72,7 @@ export class Spectral {
             opts.resolve?.documentUri,
           );
 
-    if (document.source === void 0 && opts.resolve?.documentUri !== void 0) {
+    if (document.source === null && opts.resolve?.documentUri !== void 0) {
```

Comment:
@P0lip That changes looks strange, unrelated to the gist of this PR.

Considering the code of `normalizeSource()` is 
```
const normalizeSource = (source: Optional<string>): string | null => {
  if (source === void 0) return null;
  return source && !startsWithProtocol(source) ? normalize(source) : source;
};
```

I wonder, are we also fixing an existing bug? Shouldn't this deserve a slight test?

Reply:
So actually it's not a bug. The typings are correct, but for whatever reason ts didn't complain about it.
Source can either be null or string, not undefined, so the condition was simply wrong.

Reply:
>The typings are correct, but for whatever reason ts didn't complain about it.

Weird... 👍